src/main/resources/common-services/CDAP/5.0.0/package/scripts/master.py

```diff
@@ -66,56 +66,63 @@ def killOlderProcesses(self):
-                                         Execute("rm -rf /tmp/twill_" + service + "_instances", user='zookeeper') 
-                    Execute("rm -rf /tmp/twill_master_services", user='zookeeper')                  
+        if status != 0:
+            print('fail in executing the command ls /election/master.services on zookeeper')
+            return
+        if (!(output.startswith('[') and output.endswith(']') and output == '[]')):
+            print('outt=put value is not equal to []')
+            return
+        # exectute ls /cdap/twill and save its output
+        Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " ls /twill > /tmp/twill_master_services", user='zookeeper')
+        status, output = commands.getstatusoutput("tail -n1 /tmp/twill_master_services")
+        if status != 0:
+            print('fail in executing the command ls /twill on zookeeper')
+            return
+        if (!(output.startswith('[') and output.endswith(']') and output != '[]')):
+            print('output value is equal to []')
+            return
+        print(output)
+        result = output[1:len(output)-1]
+        services = result.split(',')
+        # iterate over last output and execute ls /cdap/twill/<name>/instances and save its output
+        for service in services:
+            service = service.strip()
+            if (service != 'master.services'):
+                Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " ls /twill/" + service + "/instances > /tmp/twill_" + service + "_instances", user='zookeeper') 
+                status, output = commands.getstatusoutput("tail -n1 /tmp/twill_" + service + "_instances")
+                if status != 0:
+                    print('fail in executing the command ls /twill/' + service + '/instances')
+                    return        
+                if (!(output.startswith('[') and output.endswith(']') and output != '[]')):
+                    print('output value is equal to []')
+                    return
+                print(output)
+                result = output[1:len(output)-1]
+                # execute get /cdap/twill/<name>/instances/<last_output and save its output
+                print(params.cdap_zookeeper_quorum)
+                print(service)
+                print(result)
+                Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output", user='zookeeper')
+                status, output = commands.getstatusoutput("tail -n1 /tmp/twill_" + service + "_instances_output")
+                if status != 0:
+                    print('fail in executing the command get /twill/' + service + '/instances')  
+                    return
```

Comment:
'return' statement looks wrong here as it will skip the for look iteration for other services as well if there are multiple services to stopped. We can use 'continue' instead  to skip current service and move to next service ?. Also, we should cleanup any temporary files before moving to next service.

Reply:
Added the continue statement to continue with other services